addons/account/models/account_account.py

```diff
@@ -1064,18 +1132,109 @@ class AccountRoot(models.Model):
 
     def init(self):
         tools.drop_view_if_exists(self.env.cr, self._table)
-        self.env.cr.execute('''
+        query = SQL(
+            """
             CREATE OR REPLACE VIEW %s AS (
-            SELECT DISTINCT ASCII(code) * 1000 + ASCII(SUBSTRING(code,2,1)) AS id,
-                   LEFT(code,2) AS name,
-                   ASCII(code) AS parent_id,
-                   company_id
-            FROM account_account WHERE code != ''
-            UNION ALL
-            SELECT DISTINCT ASCII(code) AS id,
-                   LEFT(code,1) AS name,
-                   NULL::int AS parent_id,
-                   company_id
-            FROM account_account WHERE code != ''
-            )''' % (self._table,)
+                SELECT DISTINCT ASCII(code_property.value_text) * 1000 + ASCII(SUBSTRING(code_property.value_text,2,1)) AS id,
```

Comment:
It is not possible to keep a stored view.
If in one country you have non compatible (id, parent) pairs => same id with different parent; then you will end up with multiple lines having the same id.